homeassistant/bootstrap.py

```diff
@@ -109,158 +136,76 @@ def _async_setup_component(hass: core.HomeAssistant,
     if domain in hass.config.components:
         return True
 
-    setup_lock = hass.data.get('setup_lock')
-    if setup_lock is None:
-        setup_lock = hass.data['setup_lock'] = asyncio.Lock(loop=hass.loop)
+    setup_events = hass.data[DATA_SETUP]
 
-    setup_progress = hass.data.get('setup_progress')
-    if setup_progress is None:
-        setup_progress = hass.data['setup_progress'] = []
+    # is setup or in progress
+    if setup_events[domain].done():
```

Comment:
I don't think that this can happen. We only will ever add a task to `hass.data[DATA_SETUP]` once.